Indexing/CodeTree.hpp

```diff
@@ -209,11 +206,11 @@ class CodeTree
     inline ILStruct* getILS()
     {
       ASS(isLitEnd());
-      return _data<ILStruct>();
+      return reinterpret_cast<ILStruct*>(reinterpret_cast<uint64_t>(_data<ILStruct>()) & ~LIT_END);
```

Comment:
Could this be `uintptr_t` please?

Reply:
The `uint64_t` comes from `_content` as usual. But if you say it makes more sense, I obey.

Reply:
Do you think setting `DATA_BITS_START` [here](https://github.com/vprover/vampire/blob/master/Indexing/CodeTree.hpp#L236) to `INSTRUCTION_BITS_END` would work? Then, we would need to check somehow that whatever `Type` we're casting from and to satisfy `alignof(Type)>SEARCH_STRUCT` or something...

Reply:
You could, but I'm not sure why you would: that would mean that you'd have two separate pieces of data that do not share the same area of memory. Maybe I misunderstood you?

Reply:
By the way, the `getBits`/`setBits` calls [here](https://github.com/vprover/vampire/blob/master/Indexing/CodeTree.hpp#L246-L249) look useless to me, we can just read/assign `_content` directly in this case. We only need the bit-twiddling if you want to set some partial area of the bitfield.

Reply:
Yes, we would use a separate 61 bits for our data pointers but we can only do this if the data is align in such a way (which we btw don't check for all types that are used as data here). This is unproblematic for `SUCCESS` which is 0, but not so for `LIT_END` which is 2, hence the above bit magic.

Reply:
> By the way, the `getBits`/`setBits` calls [here](https://github.com/vprover/vampire/blob/master/Indexing/CodeTree.hpp#L246-L249) look useless to me [...]

I agree, but similarly [here](https://github.com/vprover/vampire/blob/master/Kernel/Term.hpp#L355) it is useless but only so because `TermTag::REF` is 0. Not saying that we should remove this though. Anyways, I can make some changes to make this more consistent for `CodeTree`.